velox/exec/tests/HashJoinBridgeTest.cpp

```diff
@@ -0,0 +1,490 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/common/file/FileSystems.h"
+#include "velox/exec/HashBuild.h"
+#include "velox/exec/HashTable.h"
+#include "velox/exec/Spill.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using facebook::velox::exec::test::TempDirectoryPath;
+
+struct TestParam {
+  int32_t numProbers{1};
+  int32_t numBuilders{1};
+};
+
+class HashJoinBridgeTest : public testing::Test,
+                           public testing::WithParamInterface<TestParam> {
+ public:
+  static std::vector<TestParam> getTestParams() {
+    return std::vector<TestParam>(
+        {TestParam{1, 4}, TestParam{4, 1}, TestParam{4, 4}, TestParam{1, 1}});
+  }
+
+ protected:
+  static void SetUpTestCase() {
+    filesystems::registerLocalFileSystem();
+  }
+
+  HashJoinBridgeTest()
+      : rowType_(ROW({"k1", "k2"}, {BIGINT(), BIGINT()})),
+        numBuilders_(GetParam().numBuilders),
+        numProbers_(GetParam().numProbers) {}
+
+  void SetUp() override {
+    rng_.seed(1245);
+    tempDir_ = exec::test::TempDirectoryPath::create();
+  }
+
+  void TearDown() override {}
+
+  uint32_t randInt(uint32_t n) {
+    std::lock_guard<std::mutex> l(mutex_);
+    return folly::Random().rand64(rng_) % (n + 1);
+  }
+
+  bool oneIn(uint32_t n) {
+    std::lock_guard<std::mutex> l(mutex_);
+    return folly::Random().oneIn(n, rng_);
+  }
+
+  std::shared_ptr<HashJoinBridge> newJoinBridge() const {
+    return std::make_shared<HashJoinBridge>();
+  }
+
+  std::unique_ptr<BaseHashTable> newFakeHashTable() {
+    std::vector<std::unique_ptr<VectorHasher>> keyHashers;
+    for (auto channel = 0; channel < rowType_->size(); ++channel) {
+      keyHashers.emplace_back(
+          std::make_unique<VectorHasher>(rowType_->childAt(channel), channel));
+    }
+    return HashTable<true>::createForJoin(
+        std::move(keyHashers), {}, true, false, mappedMemory_);
+  }
+
+  std::vector<ContinueFuture> newEmptyFutures(int32_t count) {
+    std::vector<ContinueFuture> futures;
+    futures.reserve(count);
+    for (int32_t i = 0; i < count; ++i) {
+      futures.push_back(ContinueFuture::makeEmpty());
+    }
+    return futures;
+  }
+
+  void createFakeFile(const std::string& filePath) {
```

Comment:
The file is not fake, is it? Maybe, createFile?

Why not use TempFilePath?

Reply:
Changed to createFile and I can't use TempFilePath as I want to keep it exist and let SpillFile object ton handle the file's life cycle. TempFilePath dtor will do unlink and close which might cause some exception or error condition.